### ISSUE TITLE
Windows <execinfo.h> package install error

### ISSUE THREAD

**Issue by Luke-Shaw** [ChristK/IMPACTncd_Engl#2]
Hi, 

Fascinating report and code, thanks for making public. 

I'm trying to implement the code locally myself, so that can look to run the microsimulation on a smaller population (within the BNSSG ICB). I get an error when installing `IMPACTncd_Engl_model_pkg`, code below with error followed by session info. I wonder if it is because I am on a Windows machine and it was built using a different operating system. Seems to be that a file execinfo.h does not exist when expected. 

```
> devtools::build()
── R CMD build ───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
✔  checking for file 'C:\GitHub\IMPACTncd_Engl\Rpackage\IMPACTncd_Engl_model_pkg/DESCRIPTION' (446ms)
─  preparing 'IMPACTncdEngl': (507ms)
✔  checking DESCRIPTION meta-information ...
─  cleaning src
─  installing the package to build vignettes (358ms)
         -----------------------------------
─  installing *source* package 'IMPACTncdEngl' ...
   ** using staged installation
   ** libs
   using C++ compiler: 'G__~1.EXE (GCC) 12.2.0'
   using C++11
   g++  -std=gnu++11 -I"C:/Users/LUKE~1.SHA/AppData/Local/Programs/R/R-43~1.1/include" -DNDEBUG  -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/Rcpp/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/dqrng/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/sitmo/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/BH/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/data.table/include'   -I"C:/RBuildTools/4.3/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -Wall -pedantic -fdiagnostics-color=always -c BCPEo_distribution.cpp -o BCPEo_distribution.o
   BCPEo_distribution.cpp:86:warning: noring '#pragma omp parallel-Wunknown-pragmas
      86 | #pragma omp parallel for default(shared)
         | 
   BCPEo_distribution.cpp:148:warning: noring '#pragma omp parallel-Wunknown-pragmas
     148 | #pragma omp parallel for default(shared)
         | 
   BCPEo_distribution.cpp:239:warning: noring '#pragma omp parallel-Wunknown-pragmas
     239 |   #pragma omp parallel for default(shared)
         | 
   g++  -std=gnu++11 -I"C:/Users/LUKE~1.SHA/AppData/Local/Programs/R/R-43~1.1/include" -DNDEBUG  -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/Rcpp/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/dqrng/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/sitmo/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/BH/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/data.table/include'   -I"C:/RBuildTools/4.3/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -Wall -pedantic -fdiagnostics-color=always -c BCT_distribution.cpp -o BCT_distribution.o
   BCT_distribution.cpp:71:warning: noring '#pragma omp parallel-Wunknown-pragmas]
      71 | #pragma omp parallel for default(shared)
         | 
   BCT_distribution.cpp:141:warning: noring '#pragma omp parallel-Wunknown-pragmas
     141 | #pragma omp parallel for default(shared)
         | 
   BCT_distribution.cpp:206:warning: noring '#pragma omp parallel-Wunknown-pragmas
     206 | #pragma omp parallel for default(shared)
         | 
   g++  -std=gnu++11 -I"C:/Users/LUKE~1.SHA/AppData/Local/Programs/R/R-43~1.1/include" -DNDEBUG  -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/Rcpp/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/dqrng/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/sitmo/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/BH/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/data.table/include'   -I"C:/RBuildTools/4.3/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -Wall -pedantic -fdiagnostics-color=always -c BNB_distribution.cpp -o BNB_distribution.o
   BNB_distribution.cpp:79:warning: noring '#pragma omp parallel-Wunknown-pragmas
      79 | #pragma omp parallel for default(shared)
         | 
   BNB_distribution.cpp:138:warning: noring '#pragma omp parallel-Wunknown-pragmas
     138 |   #pragma omp parallel for default(shared)
         | 
   BNB_distribution.cpp:299:warning: noring '#pragma omp parallel-Wunknown-pragmas
     299 | #pragma omp parallel for default(shared)
         | 
   BNB_distribution.cpp:335:warning: noring '#pragma omp parallel-Wunknown-pragmas
     335 | #pragma omp parallel for default(shared)
         | 
   BNB_distribution.cpp:400:warning: noring '#pragma omp parallel-Wunknown-pragmas
     400 | #pragma omp parallel for default(shared)
         | 
   BNB_distribution.cpp:468:warning: noring '#pragma omp parallel-Wunknown-pragmas
     468 | #pragma omp parallel for default(shared)
         | 
   g++  -std=gnu++11 -I"C:/Users/LUKE~1.SHA/AppData/Local/Programs/R/R-43~1.1/include" -DNDEBUG  -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/Rcpp/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/dqrng/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/sitmo/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/BH/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/data.table/include'   -I"C:/RBuildTools/4.3/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -Wall -pedantic -fdiagnostics-color=always -c DEL_distribution.cpp -o DEL_distribution.o
   DEL_distribution.cpp:118:warning: noring '#pragma omp parallel-Wunknown-pragmas
     118 | #pragma omp parallel for default(shared)
         | 
   DEL_distribution.cpp:209:warning: noring '#pragma omp parallel-Wunknown-pragmas
     209 | #pragma omp parallel for default(shared)
         | 
   DEL_distribution.cpp:273:warning: noring '#pragma omp parallel-Wunknown-pragmas
     273 | #pragma omp parallel for private(cumpro, j) shared(QQQ)
         | 
   DEL_distribution.cpp:n function 'Rcpp::NumericVector my_tofydel2(const Rcpp::NumericVector&, const Rcpp::NumericVector&, const Rcpp::NumericVector&, const Rcpp::NumericVector&)
   DEL_distribution.cpp:57:10:warning: O C++ forbids variable length array 'tofY-Wvla
      57 |   double tofYp1];
         |          ^~~~
   DEL_distribution.cpp:n function 'double my_tofydel2_scalar(const int&, const double&, const double&, const double&)
   DEL_distribution.cpp:80:10:warning: O C++ forbids variable length array 'tofY-Wvla
      80 |   double tofY2];
         |          ^~~~
   g++  -std=gnu++11 -I"C:/Users/LUKE~1.SHA/AppData/Local/Programs/R/R-43~1.1/include" -DNDEBUG  -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/Rcpp/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/dqrng/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/sitmo/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/BH/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/data.table/include'   -I"C:/RBuildTools/4.3/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -Wall -pedantic -fdiagnostics-color=always -c DPO_distribution.cpp -o DPO_distribution.o
   DPO_distribution.cpp:272:warning: noring '#pragma omp parallel-Wunknown-pragmas
     272 | #pragma omp parallel for default(shared)
         | 
   g++  -std=gnu++11 -I"C:/Users/LUKE~1.SHA/AppData/Local/Programs/R/R-43~1.1/include" -DNDEBUG  -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/Rcpp/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/dqrng/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/sitmo/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/BH/include' -I'C:/Users/Luke.Shaw/AppData/Local/Programs/R/R-4.3.1/library/data.table/include'   -I"C:/RBuildTools/4.3/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -Wall -pedantic -fdiagnostics-color=always -c IMPACTncd_sim.cpp -o IMPACTncd_sim.o
   IMPACTncd_sim.cpp:13:10:fatal error: nfo.h: No such file or directory
      13 | #include <execinfo.h>acktrace, backtrace_symbols
         |          ^~~~~~~~~~~~
   compilation terminated.
   make: *** [C:/Users/LUKE~1.SHA/AppData/Local/Programs/R/R-43~1.1/etc/x64/Makeconf:272: IMPACTncd_sim.o] Error 1
   ERROR: compilation failed for package 'IMPACTncdEngl'
─  removing 'C:/Users/LUKE~1.SHA/AppData/Local/Temp/RtmpI1Zdjj/Rinst3e1c62b447b6/IMPACTncdEngl'
         -----------------------------------
   ERROR: package installation failed
Error in `(function (command = NULL, args = character(), error_on_status = TRUE, …`:
! System command 'Rcmd.exe' failed
---
Exit status: 1
stdout & stderr: <printed>
---
Type .Last.error to see the more details.
```

```
> sessionInfo()
R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_United Kingdom.utf8  LC_CTYPE=English_United Kingdom.utf8    LC_MONETARY=English_United Kingdom.utf8 LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.utf8    

time zone: Europe/London
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] miniUI_0.1.1.1    compiler_4.3.1    crayon_1.5.2      promises_1.2.1    Rcpp_1.0.11       stringr_1.5.0     callr_3.7.3       later_1.3.1       fastmap_1.1.1     mime_0.12        
[11] R6_2.5.1          htmlwidgets_1.6.2 desc_1.4.2        profvis_0.3.8     rprojroot_2.0.3   shiny_1.7.5       rlang_1.1.1       cachem_1.0.8      stringi_1.7.12    httpuv_1.6.11    
[21] fs_1.6.3          pkgload_1.3.3     memoise_2.0.1     cli_3.6.1         magrittr_2.0.3    ps_1.7.5          digest_0.6.33     processx_3.8.2    rstudioapi_0.15.0 xtable_1.8-4     
[31] remotes_2.4.2.1   devtools_2.4.5    lifecycle_1.0.3   prettyunits_1.2.0 vctrs_0.6.3       glue_1.6.2        urlchecker_1.0.1  sessioninfo_1.2.2 pkgbuild_1.4.2    purrr_1.0.2      
[41] usethis_2.2.2     tools_4.3.1       ellipsis_0.3.2    htmltools_0.5.6  
```

**Comment by ChristK** [comment]
Dear Luke

Many thanks for your interest.

We are developing this model in Linux Ubuntu. So, the error may indeed be caused because you try to compile it on Windows. That said, we are in the process of making it compatible with Windows by the end of the year, but I do not recall seeing this error before in our tests. We are also updating the IMPACTncd_sim.cpp, which seems to complain in your case, so perhaps the error comes from that. So you can try the branch 'report_1' on Ubuntu, which we tested extensively and should work, or wait a couple of months until we iron out all the issues with Windows.

cheers
Chris

**Comment by ChristK** [comment]
Now fixed through [552f84a](https://github.com/ChristK/IMPACTncd_Engl/commit/552f84aff5af35198df5f1e31e23053d7cef89ba).
Note that the model now compiles on Windows but remains incompatible with Windows. Further changes are in the pipeline to achieve Windows compatibility.
